text/0000-gatsby-plugin-cli.md

```diff
@@ -0,0 +1,112 @@
+* remove\* - Removes npm package form `node_modules` and `package.json` and removes config from `gatsby-config.js`. potential alias: uninstall
+* disable - comments out config in `gatsby-config.js`.
+* enable - un-comments config in `gatsby-config.js`.
+* search - searches the [plugin database](https://www.gatsbyjs.org/plugins/) and returns relevant options
+
+
+\*add/remove should/could be designed to handle gatsby plugin dependencies as well.
+
+### Plugin-name
+
+* **standard**: This should directly correspond to the npm package. 
+
+* **non-standard**: For the sake of convenience it would be nice if I wanted to install (for example) `gatsby-plugin-netlify`, I could either type the full name or I could just type `plugin-netlify` or even `netlify` and the command will figure out what I mean. 
+
+In the non-standard case  though there are potential for multiple plugin results. For example, if I typed just `netlify` [6 plugins](https://www.gatsbyjs.org/plugins/?=netlify) contain `netlify`. In this case it could return something saying "I'm not sure what you mean". Even better if it choose a top option I can say yes/no to or if it gave a list number 1-6 and asked me to enter the number of the one I want to install. This being said, this I'd make this a stretch goal, nice but not required.
+
+## Config
+
+Managing the config becomes the difficult part of this imo. For plugins that don't require any configuration it is easy to simply append and object containing the name. In the case of configurable plugins the cli has to get that "default" config from somewhere.
+
+I have not spent a ton of time thinking of the *best* way to do this but my initial thought would be to add some kind of `default-config.json` in the root of plugins (alternatively maybe something that's part of the README.md). The absence or a parameter within could define whether further configuration is required beyond what is provided as well which is logged to console for the developers awareness on execution of `gatsby plugin`. For example:
+
+    $ gatsby plugin add gatsby-plugin-netlify
+    ...
+    info: 'gatsby-plugin-netlify' successfully installed and configured.
+
+    $ gatsby plugin add gatsby-plugin-netlify-cms
+    ...
+    info: 'gatsby-plugin-netlify-cms' successfully installed
+    warning: further configuration required in `gatsby-config.js`
+    info: see <https://www.gatsbyjs.org/packages/gatsby-plugin-netlify-cms/?=netlify> for more info on configuring this plugin.
+
+# Drawbacks
+
+- This should be a fairly simple implementation from a code perspective. The most difficult aspect would be updating plugins with a `default-config.json` to be compatible, but we could figure out a way to handle this for non-compliant repos that throws a warning to the dev that no default config was found so they'll have to deal with it manually and give them a link to docs.
```

Comment:
If we had a new API for this instead, the default would be to add the plugin with the shorthand:

```diff
  plugins: [
+  'gatsby-plugin-name'
  ];
```

The message could say, "No configuration options were supplied for <plugin>. See <docs> for configuration options."

That way there's nothing required for plugins that don't take config, and no regression for existing plugins — this actually improves even the default situation because now links to docs are being posted.

Reply:
yeah, this is exactally what I'm thinking, don't create a regression just through a warning to the dev to go check the docs/config.